### ISSUE TITLE
_CastError on second transaction after first call fails in beginTransaction

### ISSUE THREAD

**Issue by celestinelau** [tekartik/sqflite#611]
I encountered the following error:

```
 _CastError at SqfliteDatabaseMixin._runTransaction at package:sqflite_common/src/database_mixin.dart
at SqfliteDatabaseMixin._runTransaction (database_mixin.dart:488 )
at SqfliteDatabaseMixin.transaction.<anonymous closure> (database_mixin.dart:505 )
at SqfliteDatabaseMixin.transaction.<anonymous closure> (database_mixin.dart:504 )
at SqfliteDatabaseMixin.txnSynchronized.<anonymous closure> (database_mixin.dart:336 )
at BasicLock.synchronized (basic_lock.dart:32 )
at SqfliteDatabaseMixin.txnSynchronized (database_mixin.dart:332 )
at SqfliteDatabaseMixin.txnWriteSynchronized (database_mixin.dart:354 )
at SqfliteDatabaseMixin.transaction (database_mixin.dart:504 )
```

The relevant block of code is in _runTransaction: https://github.com/tekartik/sqflite/blob/master/sqflite_common/lib/src/database_mixin.dart#L479-L498

This occurs when `transaction()` is called multiple times, and `beginTransaction` throws an exception in the first call, leaving transactionRefCount as non-zero. Subsequently, when `_runTransaction` is called again with a null transaction, beginTransaction is not called and `result = await action(txn!);` gets invoked with a null transaction, causing the _CastError.

**Comment by alextekartik** [comment]
Good finding, I'll look into it.

**Comment by alextekartik** [comment]
Fixed in sqflite_common 1.0.3+2 /  2.0.0+2

**Comment by celestinelau** [comment]
This change fixes the issue. Thank you for the super quick fix!
